src/test/run-fail/issue-30380.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// error-pattern:panicking destructors ftw!
```

Comment:
A comment summarizing what this test case is testing would be great. I might prefer another name (but certainly link to the issue in the comments), but the best I could come up with was `panic-in-drop-during-assignment.rs`, which may not actually be particularly more helpful. It's hard to say what "category" this belongs in -- probably some series of tests we will eventually write probing execution order in various extreme situations.